demos/yolo_nas/src/demo.py

```diff
@@ -0,0 +1,164 @@
+import argparse
+from typing import List, Optional, Union
+
+import numpy as np
+import torch
+
+import norfair
+from norfair import Detection, Paths, Tracker, Video
+
+DISTANCE_THRESHOLD_BBOX: float = 0.7
+DISTANCE_THRESHOLD_CENTROID: int = 30
+MAX_DISTANCE: int = 10000
+
+
+import super_gradients
+
+
+class YOLO_NAS:
+    def __init__(self, model_name: str, device: Optional[str] = None):
+        if device is not None and "cuda" in device and not torch.cuda.is_available():
+            raise Exception(
+                "Selected device='cuda', but cuda is not available to Pytorch."
+            )
+        # automatically set device if its None
+        elif device is None:
+            device = "cuda:0" if torch.cuda.is_available() else "cpu"
+
+        # load model
+        else:
+            self.model = super_gradients.training.models.get(
+                "yolo_nas_l", pretrained_weights="coco"
+            ).cuda()
+
+    def __call__(
+        self,
+        img: Union[str, np.ndarray],
+        conf_threshold: float = 0.35,
+        iou_threshold: float = 0.45,
+        image_size: int = 720,
+        classes: Optional[List[int]] = None,
+    ) -> torch.tensor:
+
+        if classes is not None:
+            self.model.classes = classes
+
+        detections = self.model.predict(img, iou_threshold, conf_threshold)
+        return detections
+
+
+def yolo_detections_to_norfair_detections(
```

Comment:
Not something to implement in this PR but to think about and maybe add to the backlog

We've used this function so often that I believe it should be included in Norfair's utils. Maybe rename it to `xywh_detections_to_norfair` and call it with `yolo_detections.xywh`